### ISSUE TITLE
[Docs]: Next.js, where are the cache expiration settings and similar configuration

### ISSUE THREAD

**Issue by miltonlaufer** [serwist/serwist#100]
### Provide environment information

package.json: 
```JSON
{
	"dependencies": {
		"@aws-crypto/sha256-browser": "^5.2.0",
		"@babel/preset-react": "^7.23.3",
		"@fortawesome/fontawesome-svg-core": "^6.4.2",
		"@fortawesome/free-brands-svg-icons": "^6.4.2",
		"@fortawesome/free-regular-svg-icons": "^6.4.2",
		"@fortawesome/free-solid-svg-icons": "^6.4.2",
		"@fortawesome/react-fontawesome": "^0.2.0",
		"@headlessui/react": "^1.7.17",
		"@hotjar/browser": "^1.0.9",
		"@sentry/nextjs": "^7.82.0",
		"@sentry/react": "^7.82.0",
		"@sentry/tracing": "^7.82.0",
		"@serwist/next": "^8.4.4",
		"@serwist/sw": "^8.4.4",
		"@types/jest": "^29.5.11",
		"colors-convert": "^1.4.1",
		"core-js": "^3.33.3",
		"framer-motion": "^10.16.5",
		"ga-gtag": "^1.2.0",
		"intersection-observer": "^0.12.2",
		"js-base64": "^3.7.5",
		"js-xxhash": "^3.0.1",
		"lodash": "^4.17.21",
		"lodash.groupby": "^4.6.0",
		"lodash.throttle": "^4.1.1",
		"lozad": "^1.16.0",
		"lzwcompress": "^1.1.0",
		"md5": "^2.3.0",
		"mobx": "^6.12.0",
		"mobx-react": "^9.1.0",
		"mobx-state-tree": "^5.4.0",
		"next": "^14.0.3",
		"next-cookies": "^2.0.3",
		"next-themes": "^0.2.1",
		"omit-deep-lodash": "^1.1.7",
		"overlayscrollbars": "^2.4.5",
		"overlayscrollbars-react": "^0.5.3",
		"pako": "^2.1.0",
		"rc-slider": "^10.4.0",
		"react": "^18.2.0",
		"react-dom": "^18.2.0",
		"react-gtm-module": "^2.0.11",
		"react-hook-form": "^7.48.2",
		"react-hotkeys-hook": "^4.4.1",
		"react-input-slider": "^6.0.1",
		"react-query": "^3.39.3",
		"regenerator-runtime": "^0.14.0",
		"swr": "^2.2.4",
		"use-onclickoutside": "^0.4.1",
		"usehooks-ts": "^2.9.1",
		"uuid": "^9.0.1"
	},
	"devDependencies": {
		"@babel/eslint-parser": "^7.23.3",
		"@babel/preset-env": "^7.23.3",
		"@babel/preset-typescript": "^7.23.3",
		"@next/eslint-plugin-next": "^14.0.3",
		"@playwright/test": "^1.40.0",
		"@testing-library/jest-dom": "^4.2.4",
		"@testing-library/react": "^14.2.0",
		"@types/lodash.throttle": "^4.1.9",
		"@types/node": "^20.10.0",
		"@types/react": "18.2.39",
		"@types/react-dom": "18.2.17",
		"@types/react-modal": "^3.16.3",
		"@types/styled-components": "^5.1.32",
		"@types/testing-library__jest-dom": "^6.0.0",
		"@typescript-eslint/eslint-plugin": "^6.13.1",
		"@typescript-eslint/parser": "^6.13.1",
		"autoprefixer": "10.4.16",
		"babel-jest": "^29.7.0",
		"babel-plugin-transform-dynamic-import": "^2.1.0",
		"eslint": "8.54.0",
		"eslint-config-airbnb-typescript": "^17.1.0",
		"eslint-config-next": "^14.0.3",
		"eslint-config-prettier": "^9.0.0",
		"eslint-import-resolver-node": "^0.3.9",
		"eslint-plugin-eslint-comments": "^3.2.0",
		"eslint-plugin-import": "^2.29.0",
		"eslint-plugin-jsx-a11y": "^6.8.0",
		"eslint-plugin-prettier": "^5.0.1",
		"eslint-plugin-promise": "^6.1.1",
		"eslint-plugin-react": "^7.33.2",
		"eslint-plugin-react-hooks": "^4.6.0",
		"eslint-plugin-simple-import-sort": "^10.0.0",
		"eslint-plugin-unicorn": "^49.0.0",
		"eslint-plugin-unused-imports": "^3.0.0",
		"eslint-webpack-plugin": "^4.0.1",
		"husky": "^8.0.3",
		"jest": "^29.7.0",
		"jest-environment-jsdom": "^29.7.0",
		"lint-staged": "^15.1.0",
		"postcss": "8.4.31",
		"prettier": "^3.1.0",
		"sass": "^1.69.5",
		"standard-version": "^9.5.0",
		"stylelint": "^15.11.0",
		"stylelint-config-standard": "^34.0.0",
		"stylelint-scss": "^5.3.1",
		"tailwindcss": "3.3.5",
		"ts-jest": "^29.1.2",
		"turbo": "^1.10.16",
		"typedoc": "^0.25.4",
		"typedoc-plugin-markdown": "^3.17.1",
		"typescript": "^5.3.3"
	},
}
```

next.config.js:
```javascript
const withSerwist = require("@serwist/next").default({
	swSrc: "src/lib/pwa/sw.ts",
	swDest: "public/sw.js",
	scope: "/",
	maximumFileSizeToCacheInBytes: 10 * 1024 * 1024,
	reloadOnOnline: false,
})
```

sw.ts:

```javascript

installSerwist({
	precacheEntries: self.__SW_MANIFEST,
	skipWaiting: true,
	clientsClaim: true,
	navigationPreload: true,
	runtimeCaching:  [
		{
			urlPattern: /https:\/\/(cdn\..*\.ourdomain
\.com).*/,
			handler: 'CacheFirst',
			options: {}, // needed to avoid serwist crash
		},
		...defaultCache],
})
```

### Describe the problem

First of all, thank you for this library. We really appreciate your work and efforts. 

My problem: we have a project in production since 8 months, using Next 14. 
Three days ago we deployed a version that implements @Serwist/next.
We noticed a HUGE spike in server requests (20X at some point).

![2024-03-08_15-34](https://github.com/serwist/serwist/assets/3741623/3dcb0d8d-9a67-43ac-a659-a8017a0cbd2e)

CPU usage of our nodes increased 2/3X, same for out bandwidth usage. 

No particular Serwist configuration. All the images and assets (except favicon and fonts) are in on a CDN. 

### Expected behavior

We were expecting the opposite, less server load in general. 
Is there any way of reducing the requests Serwist does on the server?

**Comment by DuCanhGH** [comment]
Would you happen to have a comprehensive list of files that Serwist tries to fetch? That would help out a lot!

I'm guessing that this is due to precaching, which, by default with `@serwist/next`, caches every file in the public/ directory and every Next.js script, fetching all of them in the process. Does `exclude: [() => true]` significantly reduce the number of requests?

**Comment by miltonlaufer** [comment]
Hey @DuCanhGH , first of all, thank you for the super fast reply!

Regarding your questions:

> Would you happen to have a comprehensive list of files that Serwist tries to fetch? That would help out a lot!

Not sure if this answers it, but here is the public folder. Is that a lot for a normal Next.JS site?

```
.
├── Aleo-Regular
│   ├── Aleo-Regular-webfont.eot
│   └── Aleo-Regular-webfont.woff
├── Downloads
│   ├── mac.txt
│   └── windows.txt
├── Gilroy-Bold
│   ├── font.woff
│   └── font.woff2
├── Gilroy-ExtraBold
│   ├── font.woff
│   └── font.woff2
├── Gilroy-Light
│   ├── font.woff
│   └── font.woff2
├── Gilroy-Medium
│   ├── font.woff
│   └── font.woff2
├── Gilroy-Regular
│   ├── font.woff
│   └── font.woff2
├── Gilroy-SemiBold
│   ├── font.woff
│   └── font.woff2
├── images
│   ├── bg_scrollbar-arrow-down_hover.png
│   ├── bg_scrollbar-arrow-down.png
│   ├── bg_scrollbar-arrow-up_hover.png
│   ├── bg_scrollbar-arrow-up.png
│   ├── bg_scrollbar-track_hover.png
│   ├── bg_scrollbar-track.png
│   ├── downloading-animation.svg
│   ├── favicon.ico
│   ├── pc-logo_light.svg
│   ├── icons
│   │   ├── apple-icon-180.png
│   │   ├── pc-logo_black-border.svg
│   │   ├── pc-logo.svg
│   │   ├── fl-web.ico
│   │   ├── manifest-icon-192.maskable.png
│   │   └── manifest-icon-512.maskable.png
│   ├── knob
│   │   └── knob.png
│   ├── loading-animation.svg
│   ├── offline.jpeg
│   ├── PC-animate-in-white-preview.svg
│   ├── PC-loader-blue-on-transparent.svg
│   ├── PC-loader-white-on-tranparent.svg
│   ├── PC-loader-white-preview_6.svg
│   ├── PC-Logo-Dark-On-Light.svg
│   ├── PC-logo-full-white.svg
│   ├── plugin-icons
│   │   ├── amp.svg
│   │   ├── analysis.svg
│   │   ├── bass.svg
│   │   ├── channel-strip.svg
│   │   ├── compressor.svg
│   │   ├── creative.svg
│   │   ├── delay.svg
│   │   ├── distortion-&-saturation.svg
│   │   ├── drums.svg
│   │   ├── eq.svg
│   │   ├── filter.svg
│   │   ├── guitar.svg
│   │   ├── imaging.svg
│   │   ├── keys.svg
│   │   ├── mastering.svg
│   │   ├── modulation.svg
│   │   ├── percussion.svg
│   │   ├── pitched-percussion.svg
│   │   ├── pitch-&-time.svg
│   │   ├── reverb.svg
│   │   ├── sfx.svg
│   │   ├── strings.svg
│   │   ├── synths.svg
│   │   ├── utility.svg
│   │   ├── vocals.svg
│   │   └── wind-&-brass.svg
│   ├── plugin-mocked
│   │   ├── pluginimage.png
│   │   └── pluginlogo.png
│   ├── refer-light.svg
│   └── refer.svg
├── manifest.json
├── robots.txt
├── sw.js
└── sw.js.map
```

> Does additionalPrecacheEntries: [], exclude: [() => true] significantly reduce the number of requests?

Couldn't find `additionalPrecacheEntries` in the documentation. I just looked it up in the code, and I'll check that. Would it be too much to ask if you could share an example of _where_ and _how_ this is used? 

Thank you a lot again!

**Comment by miltonlaufer** [comment]
> Couldn't find additionalPrecacheEntries in the documentation. I just looked it up in the code, and I'll check that. Would it be too much to ask if you could share an example of where and how this is used?

BTW, this is the reason why I started this issue as a documentation one, as I think it's not a bug but a lack of understanding of how to set this right.

**Comment by DuCanhGH** [comment]
@miltonlaufer https://serwist.pages.dev/docs/build/configuring/base-options/additional-precache-entries

Here's the documentation! I guess it's a bit too buried. Maybe we can add these options to `@serwist/next`'s list of options in the sidebar :)

**Comment by miltonlaufer** [comment]
@DuCanhGH thanks again!

We finally removed the precache using `exclude` and `additionalPrecacheEntries` and we manually selected the resources we need to work offline and went back to previous levels. 

Couldn't find `exclude` in the documentation, but I checked the code and I saw the JSDocs. I noticed that the default value is not filtering "js.map" files, should I make a PR to fix that? (it worked by adding 
```javascript
const withSerwist = require("@serwist/next").default({
 //...
	exclude: [new RegExp('.*\.(js|css)\.map')],
})
```

To share a bit of history and tests:
![2024-03-11_12-47](https://github.com/serwist/serwist/assets/3741623/c30d35b1-f9f5-48b0-ba21-dd74ff57f043)

References:
1) When, hours after deploying the PWA implementation, we removed the 
2) When we removed some unused elements in `/public`
3) When we added the regex to remove the `.map` files. 
4) When we disabled the entire precache using 
```javascript
const withSerwist = require("@serwist/next").default({
 //...
	exclude: [()=>true],
	additionalPrecacheEntries: [],
})
```
